apps/analog-app/src/main-cf.server.ts

```diff
@@ -0,0 +1,38 @@
+import 'zone.js/node';
+import { InjectionToken, enableProdMode } from '@angular/core';
+import { bootstrapApplication } from '@angular/platform-browser';
+import { renderApplication } from '@angular/platform-server';
+import { APP_BASE_HREF } from '@angular/common';
+
+import { config } from './app/app.config.server';
+import { AppComponent } from './app/app.component';
+
+if (import.meta.env.PROD) {
+  enableProdMode();
+}
+
+const REQUEST = new InjectionToken<Request>('REQUEST');
+const RESPONSE = new InjectionToken<Response>('RESPONSE');
+const baseHref = process.env['CF_PAGES_URL'] ?? `http://localhost:8888`;
+
+export function bootstrap() {
+  return bootstrapApplication(AppComponent, config);
+}
+
+export default async function render(
+  url: string,
+  document: string,
+  { req, res }: { req: Request; res: Response }
+) {
+  const html = await renderApplication(bootstrap, {
+    document,
+    url: `${baseHref}${url}`,
+    platformProviders: [
+      { provide: REQUEST, useValue: req },
+      { provide: RESPONSE, useValue: res },
+      { provide: APP_BASE_HREF, useValue: baseHref },
```

Comment:
Could we please also create providers for the CF Environment and ExecutionContext objects?

@jculvey is exposing these to the to Nitro here: https://github.com/cloudflare/workers-sdk/pull/5529/files#diff-2e22ac75a5e80e245679ae59e9345d2474a6f7b5fee71841cf74bfa1a7c75922R14-R15 so it should be possible to pass these all the way into Angular's DI which would be the most natural way for Angular devs to access these objects.

Reply:
This app is just to verify the deployment works correctly. Will add an example of providing these in the docs though